arches/app/src/components/ControlledListManager/ControlledListInventory.vue

```diff
@@ -0,0 +1,196 @@
+<script setup lang="ts">
+import arches from "arches";
+import Cookies from "js-cookie";
+import { ref } from "vue";
+import { useGettext } from "vue3-gettext";
+
+import Splitter from "primevue/splitter";
+import SplitterPanel from "primevue/splitterpanel";
+import { useToast } from "primevue/usetoast";
+
+import ControlledListTable from "@/components/ControlledListManager/ControlledListTable.vue";
+import SidepanelDataView from "@/components/ControlledListManager/SidepanelDataView.vue";
+import SpinnerIcon from "@/components/SpinnerIcon.vue";
+
+import type { Ref } from "vue";
+import type {
+    ControlledList,
+    LanguageMap,
+} from "@/types/ControlledListManager.d";
+
+const props: {
+    displayedList: ControlledList | null;
+    languageMap: LanguageMap | null;
+    setDisplayedList: (list: ControlledList | null) => void;
+    setLanguageMap: (map: LanguageMap) => void;
+    setEditing: (val: boolean) => void;
+} = defineProps([
+    "displayedList",
+    "languageMap",
+    "setDisplayedList",
+    "setLanguageMap",
+    "setEditing",
+]);
+
+const items: Ref<ControlledList[]> = ref([]);
+const toast = useToast();
+const { $gettext, $ngettext } = useGettext();
+const lightGray = "#f4f4f4";
+
+const fetchLists = async () => {
+    let errorText;
+    try {
+        const response = await fetch(arches.urls.controlled_lists);
+        if (!response.ok) {
+            errorText = response.statusText;
+            const body = await response.json();
+            errorText = body.message;
+            throw new Error();
+        } else {
+            await response.json().then((data) => {
+                props.setLanguageMap(data.languages);
```

Comment:
I think `setLanguageMap` is only being used here? If so why is it defined in the parent?

Reply:
On my current understanding of state management in vue, which is pretty slight, there are a few options for child -> parent "talkback":

- emit events to a direct parent
- event events up several layers through an event bus
- defineModel
- directly mutate props (eslint not happy)
- pass a parent's ref down to the child (I recall eslint not being happy about that, or possibly reactivity breaking, but I can't recall with certainty)
- pass a setter down to the child, react-style (this instance)


I think this setLanguageMap example may be a better fit for `defineModel`, which I used elsewhere in this PR. I'll give it a go and share what I find 👍

Reply:
Actually, I can just get languages from the arches object. :bulb: 96f73082b71e2152baef5731c18505a1c8eda269